pandas/tests/plotting/test_series.py

```diff
@@ -919,10 +919,7 @@ def test_plot_order(self, data, index):
         ax = ser.plot(kind="bar")
 
         expected = ser.tolist()
-        result = [
-            patch.get_bbox().ymax
-            for patch in sorted(ax.patches, key=lambda patch: patch.get_bbox().xmax)
-        ]
+        result = [patch.get_bbox().ymax for patch in ax.patches]
```

Comment:
as is, this test expects [1,2,3,4] and subsequently [10, 50, 20, 30]. 

but on the actual displayed graph, the xticks are sorted in ascending order. to investigate what's causing this behaviour and to fix the test (or create a new one to test fixed logic)